### PR TITLE
Update Backup/Restore names

### PR DESCRIPTION
**Pull request by ryanemerson**
A bit late to the party, but without this the json for templates and tasks won't be recognised on the server.

### CODE REVIEW

**Review by rigazilla — APPROVED** [review]
lgtm, but CI is failing.

**Comment by dmvolod** [comment]
@ryanemerson I've opened issue and did some tests on it #804

**Comment by ryanemerson** [comment]
Requires https://github.com/infinispan/infinispan/pull/9034 which will be included as part of the 12.0.1.Final release

**Comment by rigazilla** [comment]
I've removed an unused import, but seems I didn't make any improvement

File: pkg/controller/zerocapacity/zero_controller.go [hunk 1]
```diff
@@ -153,10 +155,15 @@ func (z *Controller) Reconcile(request reconcile.Request) (reconcile.Result, err
 	}
 
 	phase := instance.Phase()
+
 	switch phase {
 	case "":
 		return reconcile.Result{}, instance.UpdatePhase(ZeroInitializing, nil)
 	case ZeroInitializing:
+		// Perform any transformations required on the CR for backwards-compatibility. Returning if a tranformation or error occurs
```
Comment (dmvolod) [inline pkg/controller/zerocapacity/zero_controller.go:163]: Seems to be we need to call `Transform` as early as possible. on the initial stage before `ZeroInitializing`.

**Comment by dmvolod** [comment]
@ryanemerson please pay attention, that error occurs during tests but not fails test execution
```
{"level":"error","ts":1613411305.2523618,"logger":"restore-controller","msg":"Execution failed","request.Name":"restore","error":"Unable to retrieve Restore with name 'restore' from the server","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/home/travis/gopath/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/infinispan/infinispan-operator/pkg/controller/zerocapacity.(*Controller).waitForExecutionToComplete\n\t/home/travis/gopath/src/github.com/infinispan/infinispan-operator/pkg/controller/zerocapacity/zero_controller.go:274\ngithub.com/infinispan/infinispan-operator/pkg/controller/zerocapacity.(*Controller).Reconcile\n\t/home/travis/gopath/src/github.com/infinispan/infinispan-operator/pkg/controller/zerocapacity/zero_controller.go:203\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/travis/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/travis/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/home/travis/gopath/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/home/travis/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/travis/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/home/travis/gopath/pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:88"}
```

**Comment by ryanemerson** [comment]
> @ryanemerson please pay attention, that error occurs during tests but didn't brake test execution
> 

This is caused by the new `TestBackupRestoreTransformations` test. The restore and backup errors are expected:

```
// ISPN-12675 It's not possible to create templates via rest, so the backup will fail as the template and scripts don't exist.
```

```
// ISPN-12675 The restore will fail as the backup could not complete successfully
```

**Review by dmvolod — APPROVED** [review]

File: pkg/controller/zerocapacity/zero_controller.go (hunk 1 above)
Comment (ryanemerson) [reply pkg/controller/zerocapacity/zero_controller.go:161]: Updated :+1:

File: test/e2e/backup-restore/backup_restore_test.go [hunk 2]
```diff
@@ -202,15 +290,13 @@ func assertNumEntries(cacheName, pod string, numEntries int, client ispnclient.H
 		panic(fmt.Sprintf("Unexpected response code %d", rsp.StatusCode))
 	}
 
-	/*
```
Comment (dmvolod) [inline test/e2e/backup-restore/backup_restore_test.go:205]: @ryanemerson do you need this commented code fragment?

Comment (ryanemerson) [reply test/e2e/backup-restore/backup_restore_test.go:205]: I'm not sure what you mean? In my changes I have uncommented the code.

Comment (dmvolod) [reply test/e2e/backup-restore/backup_restore_test.go:205]: Oh, sorry, was wrong to see.

**Comment by dmvolod** [comment]
Thanks, @ryanemerson . Merged.

**Pull request merged**